lib/fixed_length/parser.rb

```diff
@@ -0,0 +1,42 @@
+module FixedLength
+  class Parser
+    DEFAULT_TYPE = 'a'
+
+    attr_reader :columns, :body
+
+    def initialize(columns, body)
+      @columns = columns
+      @body = body
+    end
+
+    def parse
+      zip unpacked_body
+    end
+
+
+    private
+
+    def unpacked_body
+      body.lines.map do |line|
+        line.unpack(unpack_format)
```

Comment:
LOL it *is* cool... my issue is I feel that it obfuscates the parsing mechanism which in this case is a straightforward one.